dev/com.ibm.ws.jpa.container/src/com/ibm/ws/jpa/container/osgi/internal/JPAComponentImpl.java

```diff
@@ -205,19 +208,37 @@ protected void modified(Map<?, ?> newProperties) {
         }
     }
 
+    @Sensitive
     private void setDefaultProperties(Dictionary<String, Object> properties) {
         // Look for default integration-level properties
         Map<String, String> dProperties = new HashMap<String, String>();
         int defaultPropertiesIndex = 0;
         while (defaultPropertiesIndex >= 0) {
             String name = (String) properties.get(DEFAULT_PROPS_PREFIX + defaultPropertiesIndex + DEFAULT_PROPS_NAME);
             if (name != null) {
-                String value = (String) properties.get(DEFAULT_PROPS_PREFIX + defaultPropertiesIndex + DEFAULT_PROPS_VALUE);
-                if (name.length() > 0 && value != null && value.length() > 0) {
-                    dProperties.put(name, value);
+                // See metatype: 'ibm:obscure="true"'
+                SerializableProtectedString value = (SerializableProtectedString) properties.get(DEFAULT_PROPS_PREFIX + defaultPropertiesIndex + DEFAULT_PROPS_VALUE);
+
+                if (value != null) {
+                    if (!name.isEmpty() && !value.isEmpty()) {
+                        // Convert to java.lang.String to pass the value to the persistence providers
```

Comment:
> Converting things later would be less efficient, but potentially safer.

That was the major reason. I didnt want to have to iterate all the properties again to convert before sending to the provider.

>I was also wondering, should we be protecting this list from modification by the JPA provider?

I think it is protected. If you read through `com.ibm.ws.jpa.management.JPAPUnitInfo.createEMFactory`, we create a temporary Map of properties, then add all the integration/default properties to that Map, finally sending it to the provider via `javax.persistence.spi.PersistenceProvider.createContainerEntityManagerFactory(PersistenceUnitInfo arg0, Map arg1)`. If the provider modifies that Map in their implementation, thats fine because we are done with it after passing it in